app/models/foreman_salt/concerns/host_managed_extensions.rb

```diff
@@ -140,13 +163,11 @@ def generate_provisioning_key
         SecureRandom.hex(10)
       end
 
-      def create_salt_autosign
-        Rails.logger.info("Create salt autosign key for host #{fqdn}")
+      def configure_salt_autosign
+        Rails.logger.info("Configure salt autosign key for host #{fqdn} on #{salt_proxy.url}")
         api = ProxyAPI::Salt.new(url: salt_proxy.url)
-        key = generate_provisioning_key
+        key = salt_autosign_key
```

Comment:
```suggestion
        key = generate_salt_autosign_key
```
?

Reply:
I wanted to split the key generation and the publishing of the key. A key is generated right after the host was validated [here](https://github.com/theforeman/foreman_salt/pull/211/files#diff-179b69b42b6edeaca2c6102e7522527493004e93a18e24f539fb91d6cfdc04cdR113). In the queued orchestration task, it just uses the key of the host.
This prevents us from updating a host parameter after it has been saved.